service/history/workflow/mutable_state_impl.go

```diff
@@ -3208,7 +3210,12 @@ func (e *MutableStateImpl) AddContinueAsNewEvent(
 		timestamp.TimeValue(continueAsNewEvent.GetEventTime()),
 	)
 
-	if err = newMutableState.SetHistoryTree(ctx, newRunID); err != nil {
+	if err = newMutableState.SetHistoryTree(
+		ctx,
+		e.executionInfo.WorkflowExecutionTimeout,
+		e.executionInfo.WorkflowRunTimeout,
```

Comment:
Same here, use timeout from newMutableState after startedEvent is added on L3222.